src/ng-select/dropdown-panel.component.ts

```diff
@@ -0,0 +1,279 @@
+import {
+    Component,
+    OnDestroy,
+    Renderer2,
+    ElementRef,
+    Input,
+    EventEmitter,
+    Output,
+    ViewChild,
+    SimpleChanges,
+    NgZone,
+    TemplateRef,
+    forwardRef,
+    Inject,
+    ViewEncapsulation
+} from '@angular/core';
+
+import { VirtualScrollService } from './virtual-scroll.service';
+import { NgOption } from './ng-select.types';
+import { NgSelectComponent, DropdownPosition } from './ng-select.component';
+import { ItemsList } from './items-list';
+
+declare var process: any;
+const IS_TEST = process.env.NODE_ENV === 'TEST';
+
+@Component({
+    providers: [VirtualScrollService],
```

Comment:
Here each component instance will have separate `VirtualScrollService` instance, not sure if it is good idea